soweego/linker/classifiers.py

```diff
@@ -95,7 +96,109 @@ def prob(self, feature_vectors: pd.DataFrame) -> pd.DataFrame:
         # so we return the second column
         classifications = self.kernel.predict_proba(feature_vectors)[:, 1]
 
-        return pd.DataFrame(classifications, index=feature_vectors.index)
+        return pd.Series(classifications, index=feature_vectors.index)
+
+    def __repr__(self):
+        return f'{self.kernel}'
+
+
+class RandomForest(SKLearnAdapter, BaseClassifier):
+    """A Random Forest classifier.
+
+    This class implements :class:`sklearn.ensemble.RandomForestClassifier`.
+
+    It fits multiple decision trees on sub-samples of the dataset and
+    averages the result to get more accuracy and reduce over-fitting.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(RandomForest, self).__init__()
+
+        kwargs = {**constants.RANDOM_FOREST_PARAMS, **kwargs}
+        self.kernel = RandomForestClassifier(*args, **kwargs)
+
+    def prob(self, feature_vectors: pd.DataFrame) -> pd.DataFrame:
+        """Classify record pairs and include the probability score
+        of being a match.
+
+        :param feature_vectors: a :class:`DataFrame <pandas.DataFrame>`
+          computed via record pairs comparison. This should be
+          :meth:`recordlinkage.Compare.compute` output.
+          See :func:`extract_features() <soweego.linker.workflow.extract_features>`
+          for more details
+        :return: the classification results
+        """
+
+        match_class = self.kernel.classes_[1]
+
+        # Invalid class label
+        assert match_class == 1, (
+            f'Invalid match class label: {match_class}.'
+            'sklearn.ensemble.RandomForestClassifier.predict_proba() expects the second class '
+            'in the trained model to be 1'
+        )
+
+        # in the result, rows are classifications and columns are classes.
+        # We are in a binary setting, so 2 classes:
+        # `0` for non-matches, `1` for matches.
+        # We only need the probability of being a match,
+        # so we return the second column
+        classifications = self.kernel.predict_proba(feature_vectors)[:, 1]
+
+        return pd.Series(classifications, index=feature_vectors.index)
+
+    def __repr__(self):
+        return f'{self.kernel}'
+
+
+class VoteClassifier(SKLearnAdapter, BaseClassifier):
+    """Basic ensemble classifier which chooses the correct prediction by
+    using majority voting (aka 'hard' voting) or chooses the label which has the
+    most total probability (the argmax of the sum of predictions),
+    aka 'soft' voting.
+    """
+
+    def __init__(self, num_features, **kwargs):
+        super(VoteClassifier, self).__init__()
+        voting = kwargs.get('voting', 'soft')
```

Comment:
This default soft voting behavior should be made explicit, at least in the documentation.

Reply:
Good idea, I've created a `VOTE_CLASSIFIER_PARAMS` dict in `constants`, so that this is treated as another parameter (in commit 610e74b5c71e09c3ad5ccc3aeaab2ccd3a9e38fa )